helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -560,6 +560,51 @@ private void processMaintenanceMode(String clusterName, final boolean enabled,
     }
   }
 
+  private enum ResetPartitionFailureReason {
+    INSTANCE_NOTALIVE,
+    INSTANCE_NONEXISTENT,
+    RESOURCE_NONEXISTENT,
+    PARTITION_NONEXISTENT,
+    PARTITION_NOT_ERROR,
+    STATE_MODEL_NONEXISTENT,
+    PENDING_MSG
+  }
+
+  private String resetPartitionLogHelper(String clusterName, String instanceName,
+      String resourceName, List<String> partitionNames, String strArg,
+      ResetPartitionFailureReason reason) {
+    String errMessage = String
+        .format("Can't reset state for %s.%s on %s, because ", resourceName, partitionNames,
+            instanceName);
+
+    switch (reason) {
+      case INSTANCE_NOTALIVE:
+        errMessage +=
+            String.format("%s has never existed in cluster %s", instanceName, clusterName);
```

Comment:
nit, this is about style.
In Java, you can put fields into enum items. This format string can be a field of the ResetPartitionFailureReason enum item. Then you don't need the switch here.